Tests/scripts/utils/get_modified_files_for_testing.py

```diff
@@ -0,0 +1,194 @@
+        is_conf_json: If Tests/Conf.json has been changed.
+        sample_tests: Files to test, Like the infrastructures files.
+        modified_metadata: Metadata files.
+        is_reputations_json: If any reputation file changed.
+        is_indicator_json: If any indicator file changed.
+    """
+    sample_tests: Set[str] = set()
+    modified_metadata: Set[str] = set()
+    modified_files: Set[str] = set()
+    types_to_files: Dict[FileType, Set[str]] = create_type_to_file(git_diff)
+
+    # Checks if any common file represents in it
+    changed_common = get_common_files(
+        types_to_files.get(FileType.INTEGRATION, set()),
+        types_to_files.get(FileType.SCRIPT, set())
+    )
+
+    # Remove common files from the sets
+    for file_path in changed_common:
+        file_type = tools.find_type(file_path)
+        try:
+            types_to_files[file_type].remove(file_path)
+        except KeyError:
+            # Can be a python file that changed and now the yml representing. Will ignore
+            pass
+
+    # Sample tests are the remaining python files
+    sample_tests = sample_tests.union(types_to_files.get(FileType.PYTHON_FILE, set()))
+
+    # Modified files = YMLs of integrations, scripts and playbooks
+    modified_files = modified_files.union(
+        types_to_files.get(FileType.INTEGRATION, set()),
+        types_to_files.get(FileType.SCRIPT, set()),
+        types_to_files.get(FileType.PLAYBOOK, set()),
+    )
+
+    # Metadata packs
+    for file_path in types_to_files.get(FileType.METADATA, set()):
+        modified_metadata.add(tools.get_pack_name(file_path))
+
+    # Modified tests are test playbooks
+    modified_tests: Set[str] = types_to_files.get(FileType.TEST_PLAYBOOK, set())
+
+    # Booleans. If this kind of file is inside, its exists
+    is_conf_json = FileType.CONF_JSON in types_to_files
+
+    is_reputations_json = FileType.REPUTATION in types_to_files
+
+    is_indicator_json = FileType.INDICATOR_FIELD in types_to_files
+
+    return (
+        list(modified_files),
+        list(modified_tests),
+        list(changed_common),
+        is_conf_json,
+        list(sample_tests),
+        modified_metadata,
+        is_reputations_json,
+        is_indicator_json,
+    )
+
+
+def get_corresponding_yml_file(file_path: str) -> Optional[str]:
+    """Gets yml files from file path.
+
+    Args:
+        file_path
+
+    Returns:
+        file path of the yml file if exists else None.
+    """
+    try:
+        # Py files, Integration, script, playbook ymls
+        dir_path = os.path.dirname(file_path)
+        file_path = glob.glob(dir_path + "/*.yml")[0]
+        return file_path
+    except IndexError:  # Not matching yml - sample test
+        return None
+
+
+def get_common_files(*args: Set[str]) -> Set[str]:
+    unified_args: Set[str] = set()
+    for arg in args:
+        unified_args = unified_args.union(arg)
+    return set(arg for arg in unified_args if arg in COMMON_YML_LIST)
```

Comment:
let's pass both sets together in one list/set for example:
```
integrations_and_scripts = [types_to_files.get(FileType.INTEGRATION, set()), types_to_files.get(FileType.SCRIPT, set())]
get_common_files(integrations_and_scripts)
```
```suggestion
def get_common_files(lists: Set[str]) -> Set[str]:
    unified_integrations_and_scripts: Set[str] = set().union(*lists)
    return set(file for file in unified_integrations_and_scripts if file in COMMON_YML_LIST)
```